frontend-react/src/components/Admin/AdminReceiverDashboard.tsx

```diff
@@ -0,0 +1,775 @@
+                        Receiver:
+                    </Label>
+                    <TextInput
+                        id="input_filter_receivers"
+                        name="input_filter_receivers"
+                        type="text"
+                        autoComplete="off"
+                        aria-autocomplete="none"
+                        autoFocus
+                        onChange={(evt) => setFilterReceivers(evt.target.value)}
+                    />
+                </div>
+
+                <div className="flex-fill margin-1">
+                    <Label
+                        className="font-sans-xs usa-label"
+                        htmlFor="input_filter_errors"
+                    >
+                        Results:
+                    </Label>
+                    <TextInput
+                        id="input_filter_errors"
+                        name="input_filter_errors"
+                        type="text"
+                        autoComplete="off"
+                        aria-autocomplete="none"
+                        autoFocus
+                        onChange={(evt) =>
+                            setFilterErrorResults(evt.target.value)
+                        }
+                    />
+                </div>
+
+                <div className="flex-fill margin-1">
+                    <Label
+                        className="font-sans-xs usa-label"
+                        htmlFor="successrate-dropdown"
+                    >
+                        Success:
+                    </Label>
+                    <Dropdown
+                        id="successrate-dropdown"
+                        name="successrate-dropdown"
+                        onChange={(evt) =>
+                            setFilterRowSuccessState(
+                                (evt?.target?.value as SuccessRate) ||
+                                    SuccessRate.UNDEFINED
+                            )
+                        }
+                    >
+                        <option value={SuccessRate.UNDEFINED}>Show All</option>
+                        <option value={SuccessRate.ALL_FAILURE}>Failed</option>
+                        <option value={SuccessRate.MIXED_SUCCESS}>
+                            Mixed success
+                        </option>
+                    </Dropdown>
+                </div>
+            </form>
+            {MainRender({
+                data,
+                filterRowStatus: filterRowSuccessState,
+                datesRange: [
+                    new Date(startDate),
+                    endDate ? new Date(endDate) : new Date(),
+                ],
+                onDetailsClick: showDetailsModal,
+            })}
+
+            <Modal
+                isLarge={true}
+                className="rs-compare-modal"
+                ref={modalShowInfoRef}
+                id={"showSuccessDetails"}
+            >
+                {ModalInfoRender({ subData: currentDataForModal })}
```

Comment:
why not treat this like a React component?

Reply:
Changing to 
```html
<ModalInfoRender subData={currentDataForModal} />
```